src/D2L.Bmx/Okta/OktaApi.cs

```diff
@@ -49,41 +49,50 @@ internal class OktaApi : IOktaApi {
 	}
 
 	async Task<AuthenticateResponse> IOktaApi.AuthenticateAsync( string username, string password ) {
-		var resp = await _httpClient.PostAsJsonAsync(
-			"authn",
-			new AuthenticateRequest( username, password ),
-			SourceGenerationContext.Default.AuthenticateRequest );
-		var authnResponse = await JsonSerializer.DeserializeAsync(
-			await resp.Content.ReadAsStreamAsync(),
-			SourceGenerationContext.Default.AuthenticateResponseRaw );
-
-		if( authnResponse is {
-			SessionToken: not null,
-			Status: AuthenticationStatus.SUCCESS
-		} ) {
-			return new AuthenticateResponse.Success( authnResponse.SessionToken );
-		}
-		if( authnResponse is {
-			StateToken: not null,
-			Embedded.Factors: not null,
-			Status: AuthenticationStatus.MFA_REQUIRED
-		} ) {
-			return new AuthenticateResponse.MfaRequired(
-				authnResponse.StateToken,
-				authnResponse.Embedded.Factors
-			);
+		try {
+			var resp = await _httpClient.PostAsJsonAsync(
+				"authn",
+				new AuthenticateRequest( username, password ),
+				SourceGenerationContext.Default.AuthenticateRequest );
+			var authnResponse = await JsonSerializer.DeserializeAsync(
+				await resp.Content.ReadAsStreamAsync(),
+				SourceGenerationContext.Default.AuthenticateResponseRaw );
+
+			if( authnResponse is {
+				SessionToken: not null,
+				Status: AuthenticationStatus.SUCCESS
+			} ) {
+				return new AuthenticateResponse.Success( authnResponse.SessionToken );
+			}
+			if( authnResponse is {
+				StateToken: not null,
+				Embedded.Factors: not null,
+				Status: AuthenticationStatus.MFA_REQUIRED
+			} ) {
+				return new AuthenticateResponse.MfaRequired(
+					authnResponse.StateToken,
+					authnResponse.Embedded.Factors
+				);
+			}
+			throw new BmxException( "Okta API Authentication Response was not Success or MFA Required" );
```

Comment:
Would this text ever get shown to the user since it's caught a few lines down and then we throw a new exception with different message

Reply:
No this would not show for the user. This is more for when Debug mode is on and it shows in the inner exception to make it a bit easier to figure out what went wrong. Not sure how useful it would be tho

Reply:
Ah Oki. Just wanted to make sure it was planned to not show them

Reply:
I'd really like to avoid nesting BmxExceptions. It's intended as a mechanism to show user facing messages, and I'd like to keep it that way. Using it in multiple different ways makes our code more confusing.

Also, I'd consider throwing exception then immediately catch it below not the best form in general. There's usually a better pattern that can express the desired logic flow.

Reply:
Its hard to articulate exactly what "Okta API Authentication Response was not Success or MFA Required" exception is attempting to capture. 

There is a few situations we need to deal with here (in general):
1. Bad initial credentials
1. Failed MFA challenge
1. Bad connectivity, any other un-handled issues

Wrapping everything under one exception handler only makes sense for that third case (where its an un handled situation that could happen for any  of the **other** number of reasons).